src/ln/channelmanager.rs

```diff
@@ -1439,6 +1456,30 @@ impl ChannelManager {
 		Ok(accept_msg)
 	}
 
+	fn internal_accept_channel(&self, their_node_id: &PublicKey, msg: &msgs::AcceptChannel) -> Result<(), MsgHandleErrInternal> {
+		let (value, output_script, user_id) = {
+			let mut channel_state = self.channel_state.lock().unwrap();
+			match channel_state.by_id.get_mut(&msg.temporary_channel_id) {
+				Some(chan) => {
+					if chan.get_their_node_id() != *their_node_id {
+						return Err(MsgHandleErrInternal::disconnect_peer_with_err_msg("Got a message for a channel from the wrong node!", msg.temporary_channel_id.clone()));
```

Comment:
A node giving us message for a channel it doesn't own seems to me either a malicious node or a busted one. Disconnect it?

Reply:
I'd be ok with that, but we need to be consistent and I know we do it differently elsewhere. I think we need to come up with some vaguely consistent concept of disconnect vs just error message and I really have no idea what it should be. I was thinking earlier today there's probably situations (like maybe this) where we'd want to disconnect if they do something obnoxious but not DoS-y but only if we dont have an open channel with them. It wouldnt be worth breaking a channel over, but if they're a random peer then maybe it would be worth disconnecting. Not really sure how to handle it, but it may make sense to think about removing the distinction between Disconnect and error message for an Error type that is used in Channel/ChannelManager and then convert that to HandleError in the ChannelManager macro when returning.

In any case, all of that is obviously out-of-scope for this PR and probably only makes sense as a refactor after all the moves-into-internal-function refactors here are done. Maybe just open an issue for it, put it on the TODO list and not worry about it now?

Reply:
Yes, out-of-scope for now, done with #153